### PR TITLE
Bd 352 - Factory Contract (without timelock functionalities)

### PR DESCRIPTION
**Pull request by RiccardoGalbusera**
tests follow the structure implemented with diamond pattern, so, even if there are no conflicts, we should merge that pr first

### CODE REVIEW

**Comment by RiccardoGalbusera** [comment]
Updated with diamond pattern merge!

**Review by nicolas-racchi — APPROVED** [review]

File: test/Registry.spec.ts [hunk 1]
```diff
@@ -3,17 +3,14 @@ import { expect } from 'chai';
...
-import { Registry } from '../typechain';
-
 describe('Registry.sol', function () {
   let deployer: any;
   let user: any;
-  let registry: Registry;
-  const contractAddr1 = '0x00000000219ab540356cBB839Cbe05303d7705Fa';
-  const id1 = hre.ethers.utils.keccak256(hre.ethers.utils.toUtf8Bytes(contractAddr1));
+  let registry: Contract;
+  let IdleLiquidityModule: Contract;
```
Comment (luduvigo) [inline test/Registry.spec.ts:12]: Can we use the same naming convention here? capitalized - non-capitalized

File: contracts/Registry.sol [hunk 2]
```diff
@@ -1,32 +1,57 @@
...
+        //Begin timelock
+        modules[_id].contractAddress = _newContractAddress;
     }
 
-    modifier onlyOwner() {
-        require(msg.sender == owner, 'Only owner');
+    ///@notice Toggle global state of a module
+    ///@param _id keccak256 of module id string
+    ///@param _activated boolean to activate or deactivate module
+    function toggleModule(bytes32 _id, bool _activated) external onlyGovernance {
```
Comment (luduvigo) [inline contracts/Registry.sol:40]: Let's improve the name of this function

**Review by luduvigo — CHANGES_REQUESTED** [review]

File: contracts/Registry.sol (hunk 2 above)
Comment (RiccardoGalbusera) [reply contracts/Registry.sol:40]: renamed as swichModuleState

File: test/Registry.spec.ts [hunk 3]
```diff
@@ -23,27 +20,60 @@ describe('Registry.sol', function () {
...
+
+    const autoCompoundModuleFactory = await ethers.getContractFactory('AutoCompoundModule');
+    autoCompoundModule = (await autoCompoundModuleFactory.deploy(
+      zeroAddress, //we don't need this contract for this test
+      30
+    )) as Contract;
+    await autoCompoundModule.deployed();
   });
 
   describe('Deployment ', function () {
```
Comment (AlbyGiaco) [inline test/Registry.spec.ts:42]: Change to the name of contract

File: test/Registry.spec.ts [hunk 4]
```diff
@@ -23,27 +20,60 @@ describe('Registry.sol', function () {
...
-    it('Should assign owner at constructor', async function () {
-      expect(await registry.owner()).to.be.equal(deployer.address);
+    it('Should assign governance at constructor', async function () {
+      expect(await registry.governance()).to.be.equal(deployer.address);
     });
   });
   describe('Modules update ', function () {
```
Comment (AlbyGiaco) [inline test/Registry.spec.ts:47]: This too, change with the name of "Contract.sol - Function"

File: test/Registry.spec.ts (hunk 3 above)
Comment (AlbyGiaco) [reply test/Registry.spec.ts:42]: "Contract.sol - Function"

**Review by luduvigo — APPROVED** [review]

**Pull request merged**